spring-cloud-stream-reactive/src/main/java/org/springframework/cloud/stream/reactive/MessageChannelToInputFluxArgumentAdapter.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2016 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.stream.reactive;
+
+import reactor.core.publisher.Flux;
+
+import org.springframework.cloud.stream.annotation.Input;
+import org.springframework.cloud.stream.binding.StreamListenerArgumentAdapter;
+import org.springframework.core.MethodParameter;
+import org.springframework.core.ResolvableType;
+import org.springframework.messaging.Message;
+import org.springframework.messaging.MessageChannel;
+import org.springframework.messaging.MessageHandler;
+import org.springframework.messaging.SubscribableChannel;
+import org.springframework.messaging.converter.CompositeMessageConverter;
+
+/**
+ * Adapts an {@link org.springframework.cloud.stream.annotation.Input} annotated
+ * {@link MessageChannel} to a {@link Flux}.
+ * @author Marius Bogoevici
+ */
+public class MessageChannelToInputFluxArgumentAdapter
+		implements StreamListenerArgumentAdapter<Flux<?>, SubscribableChannel> {
+
+	private CompositeMessageConverter messageConverter;
```

Comment:
should change to one of the following:

1) switch this to a constructor arg (with an assert that it's not null)
2) set a default instance here
3) assert that it's not null when it's used below (line 76)